src/Microsoft.OpenApi.Readers/V3/OpenApiDocumentDeserializer.cs

```diff
@@ -48,12 +48,24 @@ internal static partial class OpenApiV3Deserializer
         public static OpenApiDocument LoadOpenApi(RootNode rootNode)
         {
             var openApidoc = new OpenApiDocument();
-
+            
             var openApiNode = rootNode.GetMap();
 
             ParseMap(openApiNode, openApidoc, _openApiFixedFields, _openApiPatternFields);
+            ValidatePathsField(openApidoc, rootNode);
 
             return openApidoc;
         }
+        
+        private static void ValidatePathsField(OpenApiDocument doc, RootNode rootNode)
+        {
+            var versionNode = rootNode.Find(new JsonPointer("/openapi")).GetScalarValue();
+            if (versionNode == null) return;
+            else if (versionNode.Contains("3.0") && doc.Paths == null)
```

Comment:
couldn't the path node be present but no paths defined in it? maybe we need to guard against that too?
what about version 2?
For v3.1, shouldn't we check that either webhooks or paths are present?

Reply:
Yeah these are all valid concerns that need to be addressed. Thanks for pointing that out.